File: repository-db-adapter/src/main/java/pl/mpakula/shorturl/infrastructure/UrlMappingJpa.java
package pl.mpakula.shorturl.infrastructure;


import lombok.*;

import javax.persistence.*;
import java.math.BigInteger;

@Entity
@NoArgsConstructor(access = AccessLevel.PRIVATE, force = true)
@AllArgsConstructor
@Getter
@Setter
class UrlMappingJpa {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private BigInteger id;

    @Column(unique = true)
    private final String originalUrl;

}

File: repository-db-adapter/src/main/kotlin/pl/mpakula/shorturl/infrastructure/UrlMappingJpa.kt
package pl.mpakula.shorturl.infrastructure

import java.math.BigInteger
import javax.persistence.*

@Entity
internal class UrlMappingJpa(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: BigInteger? = null,
        @Column(unique = true)
        var originalUrl: String? = null,
)